assets/data/effects/drauven/drauven_pc_cordAndVines_oneAction.json

```diff
@@ -0,0 +1,141 @@
+	],
+	"priority": "1",
+	"onlyShowIfPossibleInCharacterSheet": true,
+	"showDescriptionInTooltip": true
+},
+"targets": [
+	{
+		"template": "SELF",
+		"cost": { "class": "ActionPoints" },
+		"missionFeedback": { "orientTo": "target" },
+		"aspects": [ "alive", "mysticDeck_mythweaver", "SELF" ]
+	},
+	{
+		"role": "source",
+		"template": "ANY",
+		"type": "SCENERY",
+		"choose": "BY_SCORE",
+		"scoreFunction": "plant+textile",
+		"scoreThreshold": "1",
+		"missionFeedback": {
+			"suggestionScript": "figures.abilitySelectOption",
+			"hoverScript": "figures.abilitySelectHover",
+			"selectedScript": "figures.abilitySelectSelected",
+			"rangeFeedback": "FRIENDLY_EFFECT_RANGE",
+			"suggestionFeedback": "FRIENDLY_EFFECT_SUGGESTION",
+			"hoverFeedback": "FRIENDLY_EFFECT_HOVER"
+		},
+		"relationship": "interfusedWithEntity"
+	},
+	{
+		"template": "ADJACENT_ENEMY",
+		"showAnyway": true,
+		"alwaysSelect": true,
+		"aspects": [ "HOSTILE_TARGET", "alive" ],
+		"relativeTo": "source",
+		"range": "(4.6+self.BONUS_RANGE)+self.mysticDeck_naturalist",
+		"lineOfSight": "LIVE_FIGURES"
+	},
+	{
+		"role": "targetTile",
+		"template": "TILE",
+		"showAnyway": true,
+		"alwaysSelect": true,
+		"missionFeedback": {
+			"rangeFeedback": "FRIENDLY_EFFECT_RANGE",
+			"suggestionFeedback": "FRIENDLY_EFFECT_SUGGESTION",
+			"hoverFeedback": "FRIENDLY_EFFECT_HOVER",
+			"applyRoleFootprint": "target",
+			"showMovementFeedback": true
+		},
+		"relativeTo": "target",
+		"range": "(3.1+self.BONUS_RANGE)+self.mysticDeck_naturalist",
+		"lineOfSight": "TERRAIN_ONLY",
+		"tileFilter": "validMoveEnd",
+		"roleMustFit": "target",
+		"footprintWithinRange": "target",
+		"footprintWithinRangeCompletely": true
+	}
+],
+"outcomes": [
+	{
+		"class": "AttackRoll",
+		"intermediateRole": "source",
+		"attackAnimationType": "magic",
+		"particles": [
+			{ "script": "cordAndVines", "setPosition": "source", "setPosition2": "target" }
+		],
+		"audioOverride": { "attackHitCue": "Characters/Default/Ability_InterfusePlant" },
+		"flankMarkOrigin": "source",
+		"defenseRollTags": [ "range" ],
+		"attackStrength": "self.RANGE_ACCURACY+10",
+		"wield": "light",
+		"always": { "class": "Damage", "target": "source", "amount": "1", "showInTooltip": false },
+		"onHit": {
+			"class": "DoAll",
+			"outcomes": [
+				{
+					"class": "Test",
+					"STUB": "If the target isn't immuneToKnockback, move them!",
+					"value": "target.immuneToKnockback",
+					"threshold": "1",
+					"onFail": { "class": "MoveEntity", "directlyToTile": "targetTile" }
+				},
+				{
+					"class": "Damage",
+					"amount": "(1d3+self.POTENCY)+self.SPELL_DAMAGE",
+					"useEquippedWeaponForStunt": true
+				},
+				{
+					"class": "Test",
+					"value": "self.mysticDeck_naturalist",
+					"threshold": "1",
+					"onPass": {
+						"class": "DoAll",
+						"outcomes": [
+							{
+								"class": "Aspects",
+								"addAspects": [
+									{ "id": "pinned", "value": "2+self.mysticDeck_mythweaver_upgrade" }
```

Comment:
Is the pinned effect supposed to only be for the upgrade? If so, we might need to swap the `+` to `*`?

Reply:
It's a hobble normally, the test above swaps it to a pin.  The Mythweaver upgrade just enhances the duration, no matter which effect is being applied.

Reply:
OH! I missed the wrapping `test` condition! Yeah, that makes a lot more sense now